test-infra/ansible/roles/node-identifier/files/identify_and_register.sh

```
@@ -0,0 +1,94 @@
+    # PCIe Link Speed (LnkSta)
+    # sudo lspci -vv -s ... requires privileges
+    PCIE_SPEED=$(lspci -vv -s "$BUS_INFO" 2>/dev/null | grep LnkSta: | grep -o 'Speed [^,]*' | head -1 | awk '{print $2}')
+else
+    MODEL="Unknown"
+    PCIE_SPEED="Unknown"
+fi
+
+# Sanitize Labels (replace spaces with dashes, etc)
+# GitHub labels: alphanumeric, -_., start/end with alphanumeric.
+sanitize() {
+    echo "$1" | sed 's/[^a-zA-Z0-9._-]/-/g'
```

Shellcheck (0.11.0)

[style] See if you can use ${variable//search/replace} instead.

(SC2001)